manifests/install.pp

```diff
@@ -14,86 +14,81 @@
     if $puppet_agent::aio_upgrade_required {
-      $install_script = 'osx_install.sh.erb'
-
-      $_logfile = "${::env_temp_variable}/osx_install.log"
-      notice("Puppet install log file at ${_logfile}")
-
-      $_installsh = "${::env_temp_variable}/osx_install.sh"
-      file { "${_installsh}":
-        ensure  => file,
-        mode    => '0755',
-        content => template('puppet_agent/do_install.sh.erb')
-      }
-      -> exec { 'osx_install script':
-        command => "${_installsh} ${::puppet_agent_pid} 2>&1 > ${_logfile} &",
+      class { 'puppet_agent::install::darwin':
+        package_version => $package_version,
+        install_options => $install_options,
       }
+      contain '::puppet_agent::install::darwin'
     }
   } elsif $::osfamily == 'windows' {
     # Prevent re-running the batch install
     if $puppet_agent::aio_upgrade_required {
-      class { 'puppet_agent::windows::install':
+      class { 'puppet_agent::install::windows':
         install_dir     => $install_dir,
         install_options => $install_options,
       }
-    }
-  } elsif ($::osfamily == 'Debian') and ($package_version != 'present') {
-    package { $::puppet_agent::package_name:
-      ensure          => "${package_version}-1${::lsbdistcodename}",
-      install_options => $install_options,
+      contain '::puppet_agent::install::windows'
     }
   } else {
+    if $::operatingsystem == 'AIX' {
+      # AIX installations always use RPM directly since no there isn't any default package manager for rpms
+      $_package_version = $package_version
+      $_install_options = concat(['--ignoreos'],$install_options)
+      $_provider = 'rpm'
+      $_source = "${::puppet_agent::params::local_packages_dir}/${::puppet_agent::prepare::package::package_file_name}"
+    } elsif $::osfamily == 'Debian' {
+      $_install_options = $install_options
+      if $::puppet_agent::absolute_source {
+        # absolute_source means we use dpkg on debian based platforms
+        $_package_version = 'present'
+        $_provider = 'dpkg'
+        $_source = "${::puppet_agent::params::local_packages_dir}/${::puppet_agent::prepare::package::package_file_name}"
+      } else {
+        # any other type of source means we use apt with no 'source' defined in the package resource below
+        if $package_version == 'present' {
+          $_package_version = $package_version
+        } else {
+          $_package_version = "${package_version}-1${::lsbdistcodename}"
+        }
+        $_provider = 'apt'
+        $_source = undef
+      }
+    } else { # RPM platforms: EL and SUSE
+      $_install_options = $install_options
+      if $::puppet_agent::absolute_source {
+        # absolute_source means we use rpm on EL/suse based platforms
+        $_package_version = 'present'
+        $_provider = 'rpm'
+        $_source = "${::puppet_agent::params::local_packages_dir}/${::puppet_agent::prepare::package::package_file_name}"
```

Comment:
does the same comment from line 57 re: absolute source apply here as well?

Reply:
Yeah, I'll need to update the prepare steps for these platforms to use prepare::package